src/ChainSafe.Gaming.HyperPlay/HyperPlayTransactionExecutor.cs

```diff
@@ -0,0 +1,67 @@
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using ChainSafe.Gaming.Evm.Signers;
+using ChainSafe.Gaming.Evm.Transactions;
+using ChainSafe.Gaming.Web3;
+using ChainSafe.Gaming.Web3.Core.Evm;
+using Nethereum.RPC.Eth.DTOs;
+
+namespace ChainSafe.Gaming.HyperPlay
+{
+    /// <summary>
+    /// Concrete implementation of <see cref="ITransactionExecutor"/> via HyperPlay desktop client.
+    /// </summary>
+    public class HyperPlayTransactionExecutor : ITransactionExecutor
+    {
+        private readonly IHyperPlayProvider hyperPlayProvider;
+        private readonly ISigner signer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HyperPlayTransactionExecutor"/> class.
+        /// </summary>
+        /// <param name="hyperPlayProvider">HyperPlay provider for making RPC requests.</param>
+        /// <param name="signer">Signer for fetching public address.</param>
+        public HyperPlayTransactionExecutor(IHyperPlayProvider hyperPlayProvider, ISigner signer)
+        {
+            this.hyperPlayProvider = hyperPlayProvider;
+            this.signer = signer;
+        }
+
+        /// <summary>
+        /// Send a Transaction via HyperPlay desktop Client.
+        /// This prompts user to approve a transaction on HyperPlay.
+        /// </summary>
+        /// <param name="transaction">Transaction to send.</param>
+        /// <returns>Hash response of a successfully executed transaction.</returns>
+        /// <exception cref="Web3Exception">Throws Exception if executing transaction fails.</exception>
+        public async Task<TransactionResponse> SendTransaction(TransactionRequest transaction)
+        {
+            if (string.IsNullOrEmpty(transaction.From))
+            {
+                transaction.From = signer.PublicAddress;
+            }
+
+            TransactionInput transactionInput = new TransactionInput
+            {
+                From = transaction.From,
+                To = transaction.To,
+                Gas = transaction.GasLimit,
+                GasPrice = transaction.GasPrice,
+                Value = transaction.Value,
+                Data = transaction.Data ?? "0x",
+                Nonce = transaction.Nonce,
+                AccessList = transaction.AccessList,
+            };
+
+            string hash = await hyperPlayProvider.Request<string>("eth_sendTransaction", transactionInput);
+
+            string hashPattern = @"^0x[a-fA-F0-9]{64}$";
+            if (!Regex.IsMatch(hash, hashPattern))
+            {
+                throw new Web3Exception($"incorrect txn response format {hash}");
+            }
+
+            return await hyperPlayProvider.Request<TransactionResponse>("eth_getTransactionByHash", hash);
```

Comment:
Don't know if this is applicable for the HyperPlayer provider, but we do polling with timeout for all the other TxExecutors (except MM as I just found out) for this call. The reason is, that it usually takes time for a transaction to be registered on a node, so when you try to get it by hash it fails at first.

Look at RpcProviderExtensions.GetTransaction() for more context.

Reply:
yep, I'll test it a bit more to make sure

Reply:
it seems to work consistently, maybe HyperPlay does their own polling until a node is confirmed

Reply:
if it works, then no, keep it as it is